daemon/policy.go

```
@@ -302,6 +308,86 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (newRev uint
 	return newRev, nil
 }
 
+type policyAddQueueOptions struct {
+	rules   policyAPI.Rules
+	opts    *AddOptions
+	context *policyQueueContext
+	reFetch bool
+}
+
+type policyQueueContext struct {
+	policyRevision uint64
+	err            error
+	done           chan bool
+	totalTime      spanstat.SpanStat
+	queueTime      spanstat.SpanStat
+}
+
+// policyAddToQueue adds the request to policyAdd in the queue and return a
+// context that will be updated when the done channel is closed.
+func (d *Daemon) policyAddToQueue(rules policyAPI.Rules, opts *AddOptions, reFetch bool) *policyQueueContext {
```

Please return the error separately as we do in all other functions